crates/core/src/test_utils/deploy_helpers.rs

```diff
@@ -867,15 +874,11 @@ impl KakarotTestEnvironmentContext {
         (&self.kakarot_client, &self.kakarot)
     }
 
-    pub fn resources_with_contract(
+    pub fn resources_and_contract(
         &self,
         contract_name: &str,
-    ) -> (&KakarotClient<JsonRpcClient<HttpTransport>>, &DeployedKakarot, &Contract, Address) {
-        let contract = self.evm_contract(contract_name);
-        let eth_address: Felt252Wrapper = contract.addresses.eth_address.into();
-        let contract_eth_address: Address = eth_address.try_into().expect("Failed to convert address");
-
-        (&self.kakarot_client, &self.kakarot, contract, contract_eth_address)
+    ) -> (&KakarotClient<JsonRpcClient<HttpTransport>>, &DeployedKakarot, &Contract) {
+        (&self.kakarot_client, &self.kakarot, self.evm_contract(contract_name))
```

Comment:
I don't understand in which circumstance it can be valuable to have this function returning `&self.kakarot_client, &self.kakarot`, given that you probably end up having

```
let (kakarot_client, kakarot, contract) = env.resources_and_contract(contract_name)
```

with

```
kakarot_client = env.kakarot_client
kakarot = env.kakarot
```

Reply:
So mainly the reason why we created these two functions in this [pr](https://github.com/kkrt-labs/kakarot-rpc/pull/399/files) (`resources` and `resources_with_contract`) was to remove a lot of boilerplate code. There are combinations of the resources that we want, sometimes not entirely using them i.e.

`let (client, _, _, plain_opcodes_eth_address) = kakarot_test_env_ctx.resources_with_contract("PlainOpcodes");`

so I thought to create these other functions so we only get what we need. Open to discuss what would be a better approach for this and what the tradeoffs are

Reply:
you don't really answer the question of why it is better to do

```
let (kakarot_client, kakarot, contract) = env.resources_and_contract(contract_name)
```

instead of

```
let contract = env.resources_and_contract(contract_name)
let kakarot_client = env.kakarot_client
let kakarot = env.kakarot
```

given the fact the two last lines 

```
let kakarot_client = env.kakarot_client
let kakarot = env.kakarot
```

don't seem really needed neither

Reply:
It would be to reduce the boiler plate code that we do for the tests, for example:

https://github.com/kkrt-labs/kakarot-rpc/blob/0f4a260c8e8e77f637b0930052015f34c87ce533/crates/core/tests/client.rs#L110-L138

we were doing this before: 

```rust
        let erc20 = test_environment.evm_contract("ERC20");
        let erc20_eth_address = {
            let address: Felt252Wrapper = erc20.addresses.eth_address.into();
            address.try_into().unwrap()
        };
        let client = test_environment.client();
        let kakarot = test_environment.kakarot();
```

And so I thought to make this simpler and just get everything in one line

Reply:
I have to agree with Clement here, I like having things super simple and I don't see the problem with accessing test_environment getters when needed

I'd suggest removing these multi-getters helpers, they save lines but bundle things together for no good reason

We can do it in a later PR, and close this PR. I'd suggest creating an issue to delete these helpers and go back to simple inlined getters for when we need them

It'll simplify everything

wdyt @ftupas @ClementWalter